internal/versions/versions.go

```diff
@@ -249,64 +310,51 @@ func (h *handler) AddReleaseImage(releaseImageUrl, pullSecret, ocpReleaseVersion
 		}
 	}
 
-	// Get minor version
-	ocpVersionKey, err := h.getKey(ocpReleaseVersion)
-	if err != nil {
-		return nil, err
-	}
-
 	// Ensure a relevant OsImage exists
-	osImage, err := h.GetOsImage(ocpVersionKey, cpuArchitecture)
+	osImage, err := h.GetOsImage(ocpReleaseVersion, cpuArchitecture)
 	if err != nil || osImage.URL == nil {
-		return nil, errors.Errorf("No OS images are available for version: %s", ocpVersionKey)
+		return nil, errors.Errorf("No OS images are available for version: %s", ocpReleaseVersion)
 	}
 
 	// Fetch ReleaseImage if exists
-	releaseImage, err := h.GetReleaseImage(ocpVersionKey, cpuArchitecture)
-	if err != nil {
+	releaseImage := funk.Find(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		return *releaseImage.OpenshiftVersion == ocpReleaseVersion && *releaseImage.CPUArchitecture == cpuArchitecture
+	})
+	if releaseImage == nil {
 		// Create a new ReleaseImage
 		releaseImage = &models.ReleaseImage{
-			OpenshiftVersion: &ocpVersionKey,
+			OpenshiftVersion: &ocpReleaseVersion,
 			CPUArchitecture:  &cpuArchitecture,
+			URL:              &releaseImageUrl,
+			Version:          &ocpReleaseVersion,
 		}
-	} else {
-		// Remove original ReleaseImage from array
-		h.releaseImages = funk.Subtract(h.releaseImages, models.ReleaseImages{releaseImage}).([]*models.ReleaseImage)
-	}
-
-	// Update ReleaseImage
-	releaseImage.URL = &releaseImageUrl
-	releaseImage.Version = &ocpReleaseVersion
 
-	// Store in releaseImages array
-	h.releaseImages = append(h.releaseImages, releaseImage)
-	h.log.Infof("Stored release version: %s", ocpReleaseVersion)
+		// Store in releaseImages array
+		h.releaseImages = append(h.releaseImages, releaseImage.(*models.ReleaseImage))
+		h.log.Infof("Stored release version: %s", ocpReleaseVersion)
+	}
 
-	return releaseImage, nil
+	return releaseImage.(*models.ReleaseImage), nil
 }
 
 // Get CPU architectures available for the specified openshift version
 // according to the OS images list.
-func (h *handler) GetCPUArchitectures(openshiftVersion string) ([]string, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
-	}
-	cpuArchitectures := []string{}
-	for _, osImage := range h.osImages {
-		if *osImage.OpenshiftVersion == versionKey {
-			if swag.StringValue(osImage.CPUArchitecture) == "" {
-				// Empty or missing property implies default CPU architecture
-				defaultArch := common.DefaultCPUArchitecture
-				osImage.CPUArchitecture = &defaultArch
-			}
-			if !funk.Contains(cpuArchitectures, *osImage.CPUArchitecture) {
-				cpuArchitectures = append(cpuArchitectures, *osImage.CPUArchitecture)
-			}
+func (h *handler) GetCPUArchitectures(openshiftVersion string) []string {
+	// Find OS images of specified openshiftVersion
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
```

Comment:
In this case, I don't think two loops through the list helps make this easier to read.
Maybe that made sense in the earlier function where you would have to check in multiple places, but I think the previous version of this function was fine.